python/lsst/meas/extensions/scarlet/io.py

```diff
@@ -472,3 +482,153 @@ def oldScarletToData(blend: Blend, psfCenter: tuple[int, int], xy0: Point2I):
+    as arguments, as well as a `from_zip` method that accepts a dictionary
+    of files to create the object from the zip archive.
+    """
+
+    default_extension = ".scarlet"
+    unsupported_parameters = None
+    can_read_from_uri = True
+
+    def _extract_all_files(self, uri: ResourcePath) -> dict[str, Any]:
+        """
+        Extract all files from a ZIP archive.
+
+        Parameters
+        ----------
+        uri : `ResourcePath`
+            URI of the ZIP file
+
+        Returns
+        -------
+        files :
+            Dictionary mapping filenames to the desired file type.
+        """
+        result = {}
+        with uri.open("rb") as f:
+            with zipfile.ZipFile(f, 'r') as zipf:
+                for filename in zipf.namelist():
+                    with zipf.open(filename) as f:
+                        result[filename] = from_json(f.read())
+        return result
+
+    # Extracting a single file from the ZIP
+    def _extract_blend(self, uri: ResourcePath, blend_filename: str) -> dict[str, Any]:
+        """Extract a single file from a ZIP archive.
+
+        Parameters
+        ----------
+        uri : `ResourcePath`
+            URI of the ZIP file
+        blend_filename : `str`
+            Name of the blend file to extract
+
+        Returns
+        -------
+        blend :
+            Dictionary mapping filenames to the desired file type.
+        """
+        results = {}
+        filenames = [
+            blend_filename,
+            'psf.json',
+            'psf_shape.json',
+        ]
+        with uri.open("rb") as f:
+            with zipfile.ZipFile(f, 'r') as zipf:
+                for filename in filenames:
+                    with zipf.open(filename) as f:
+                        results[filename] = from_json(f.read())
+        return results
+
+    def _build_model(self, zip_dict: dict[str, Any]) -> scl.io.ScarletModelData:
+        """Build a ScarletModelData instance from a dictionary of files.
+
+        Parameters
+        ----------
+        zip_dict : `dict` of `str` to `object`
+            Dictionary mapping filenames to the desired file type.
+
+        Returns
+        -------
+        model :
+            ScarletModelData instance.
+        """
+        model_psf = zip_dict['psf.json']
+        psf_shape = zip_dict['psf_shape.json']
+        blends = {}
+        for key, value in zip_dict.items():
+            if key.startswith('blend_'):
+                blend_id = int(key.split('.')[0].split('_')[1])
+                blends[blend_id] = value
+        return scl.io.ScarletModelData.parse_obj({
+            'psf': model_psf,
+            'psfShape': psf_shape,
+            'blends': blends,
+        })
+
+    def _model_to_zip(self, model_data: scl.io.ScarletModelData) -> dict[str, Any]:
+        """Convert a ScarletModelData instance to a dictionary of files.
+
+        Parameters
+        ----------
+        model_data : `scl.io.ScarletModelData`
```

Comment:
Is this in fact `lsst.scarlet.lite.io.ScarletModelData`? Please use the full name since sphinx does not understand import aliases. Similarly it can't work out what `ResourcePath` means on its own.